pkgs/development/compilers/rust/rustfmt.nix

```diff
@@ -21,6 +21,8 @@ rustPlatform.buildRustPackage rec {
   CFG_RELEASE = rustPlatform.rust.rustc.version;
   CFG_RELEASE_CHANNEL = "stable";
 
+  patches = [ ./rustfmt-fix-self-tests.patch ];
```

Comment:
Can we use `fetchpatch` here rather than committing the patch?

Reply:
Not really, upstream rustc just does big merges from the rustfmt repo from time to time, so there's no individual commit with a fix we can pick.

Reply:
Should be able to get it from `rustfmt` and adjust it to match with `extraPrefix`.

Reply:
https://github.com/rust-lang/rustfmt/commit/89ca3f3a100456d652b156b1a62b8e244e526c4e

What is even.

Reply:
It's a pretty basic change. I'm not sure its worth the effort to do that

Reply:
Okay, upstream patch is no good. Please add a comment instead. "fixed upstream, can be removed next release" or similar.